Update to secondary indexes doc (#4588)
timescale/docs: use-timescale/hypercore/secondary-indexes.md

For example, to access information about a single device with a specific `device_id`, you segment on the `device_id` column. 
This enables you to run analytical queries on compressed data in the $COLUMNSTORE much faster.

> To illustrate, let's create a $HYPERTABLE and then run the same query on it with and without optimizations:

Reviewer: I don't think this will work. Columnstore is enabled by default from 2.23 on.

> To illustrate, let's create a $HYPERTABLE and then run the same query on it with and without optimizations:

Reviewer: ```suggestion
To illustrate, run the same query on a $HYPERTABLE, first without, then with optimizations:
```

<Procedure>

1. **Create a $HYPERTABLE**

   Create a `metrics` $HYPERTABLE with the following command:  

    ```sql
    CREATE TABLE metrics (
      time TIMESTAMPTZ,
      user_id INT,
      device_id INT,
      data JSONB
    ) WITH (
      tsdb.hypertable
    );
    ```

   <CreateHypertablePolicyNote />

1. **Execute a query on the $HYPERTABLE without optimizations**

   1. Query your data
      ```sql
      SELECT device_id, AVG(cpu) AS avg_cpu, AVG(disk_io) AS avg_disk_io
      FROM metrics
      WHERE time >= '2024-03-01 00:00:00+01'
        AND time < '2024-03-02 00:00:00+01'
        AND device_id = 5
      GROUP BY device_id;
      ```
      Gives the following result:
      ```sql
      device_id |      avg_cpu       |     avg_disk_io
      -----------+--------------------+---------------------
      5 | 0.4954351575883885 | 0.49725603413909114
      (1 row)
      Time: 29.216 ms
      ```

1. **Execute a query on the same data segmented and ordered in the $COLUMNSTORE**

   1. Control the way your data is ordered and segmented in the $COLUMNSTORE:

      ```sql
      ALTER TABLE metrics SET (

>         timescaledb.enable_columnstore = true,

Reviewer: Columnstore is already enabled, you don't need to enable it.
